### PR TITLE
TASK: Adjust requests and responses to Flow 6 PSR-7 interfaces

### PR DESCRIPTION
**Pull request by daniellienert**

### CODE REVIEW

File: Classes/Transfer/Exception.php [hunk 1]
```diff
@@ -10,35 +12,27 @@
...
-     * Exception constructor.
-     *
-     * @param string $message
-     * @param int $code
-     * @param FlowResponse $response
-     * @param Request $request
-     * @param \Exception $previous
-     */
-    public function __construct($message, $code, FlowResponse $response, Request $request = null, \Exception $previous = null)
+    public function __construct($message, $code, ResponseInterface $response, RequestInterface $request = null, \Exception $previous = null)
```
Comment (kdambekalns) [inline Classes/Transfer/Exception.php]: Not sure about the dropped docblock – two arguments are now without type hint and in theory we still require docblocks in our CGL.

File: Classes/Domain/Model/Index.php [hunk 2]
```diff
@@ -152,14 +155,14 @@ public function exists()
...
      * @param array $arguments
      * @param string $content
-     * @param boolean $prefixIndex
+     * @param bool $prefixIndex
      * @return Response
-     * @throws ElasticSearchException
+     * @throws \Exception
      */
-    public function request($method, $path = null, array $arguments = [], $content = null, $prefixIndex = true)
+    public function request(string $method, ?string $path = null, array $arguments = [], ?string $content = null, bool $prefixIndex = true): Response
```
Comment (kdambekalns) [inline Classes/Domain/Model/Index.php]: `?string $foo = null` is not needed, if `null` is given as default value, the "nullable modifier" is not needed. And vice versa, AFAIK.

File: Classes/Domain/Model/Index.php [hunk 3]
```diff
@@ -152,14 +155,14 @@ public function exists()
      * @param string $path
      * @param array $arguments
      * @param string $content
-     * @param boolean $prefixIndex
+     * @param bool $prefixIndex
      * @return Response
-     * @throws ElasticSearchException
+     * @throws \Exception
```
Comment (kdambekalns) [inline Classes/Domain/Model/Index.php]: Wait, why? What is wrong with `ElasticSearchException`? It's imported above…

File: Classes/Domain/Model/Index.php [hunk 4]
```diff
@@ -152,14 +155,14 @@ public function exists()
...
      */
-    public function request($method, $path = null, array $arguments = [], $content = null, $prefixIndex = true)
+    public function request(string $method, ?string $path = null, array $arguments = [], ?string $content = null, bool $prefixIndex = true): Response
     {
         if ($this->client === null) {
-            throw new Exception('The client of the index "' . $this->name . '" is not set, hence no requests can be done.');
+            throw new \Exception('The client of the index "' . $this->name . '" is not set, hence no requests can be done.', 1566313883);
```
Comment (kdambekalns) [inline Classes/Domain/Model/Index.php]: Ok, so the `ElasticSearchException` that was advertised was not thrown. But it's still the same (only aliased). Now `\Exception` is something else, though!

File: Classes/Domain/Model/Index.php [hunk 5]
```diff
@@ -173,8 +176,9 @@ public function request($method, $path = null, array $arguments = [], $content =
 
     /**
      * @return void
+     * @throws \Exception
```
Comment (kdambekalns) [inline Classes/Domain/Model/Index.php]: Rather generic, and… is it correct?

File: Classes/Domain/Model/Index.php [hunk 6]
```diff
@@ -196,6 +200,7 @@ protected function getSettings()
...
```
Comment (kdambekalns) [inline Classes/Domain/Model/Index.php]: Rather generic, and… is it correct?

File: Classes/Domain/Model/Index.php (hunk 4 above)
Comment (daniellienert) [reply Classes/Domain/Model/Index.php]: Thanks!

File: Classes/Domain/Model/Index.php (hunk 5 above)
Comment (daniellienert) [reply Classes/Domain/Model/Index.php]: Nope

**Comment by daniellienert** [comment]
Thanks for the Review @kdambekalns. Please have another look

**Pull request merged**